pipeline/rds_to_s3.py

```diff
@@ -0,0 +1,123 @@
+        query = db.select(table).where(table.columns.datetime < str.encode(date_cutoff))
+        response = connection.execute(query)
+        return pd.DataFrame(response.fetchall())
+
+    except Exception as e:
+        raise e
+
+
+def get_day_bucket_keys(s3_client: client, folder_path: str, day: int = TODAY.day,
+                    bucket_name: str = environ['BUCKET_NAME']) -> list:
+    """
+    Returns list of keys within a given s3 bucket ending in '_{day}.csv, with a prefix matching
+    the given folder_path.
+    """
+    objects = s3_client.list_objects(Bucket=bucket_name, Prefix=folder_path).get('Contents')
+    if objects:
+        return [obj['Key'] for obj in objects if obj['Key'].split('_')[-1] == f'{day}.csv']
+    return []
+
+
+def get_current_csv_data(data_type: str, s3_client: client, bucket_name:
+                         str = environ['BUCKET_NAME'], date: str = TODAY) -> pd.DataFrame:
+    """
+    Downloads relevant files of specified data_type (watering/recording) from S3 to local, and
+    then returns it as a pandas dataframe
+    """
+
+    folder_path = f'{date.year}/{date.month}'
+    keys = get_day_bucket_keys(s3_client, folder_path)
+    type_keys = [key for key in keys if data_type in key]
+    if type_keys:
+        type_key = type_keys[0]
+        response = s3_client.get_object(Bucket=bucket_name, Key = type_key)
+        return pd.read_csv(response.get("Body"))
+    return pd.DataFrame()
+
+
+def upload_to_s3(data_type: str, df: pd.DataFrame, s3_client,
+                 bucket_name: str = environ['BUCKET_NAME'], date: str = TODAY):
+    """Uploads pandas dataframe of data_type to appropriate csv in s3 bucket."""
+    s3_client.put_object(Body = df.to_csv(), Bucket = bucket_name,
+                         Key = f'{date.year}/{date.month}/{data_type}_{date.day}.csv')
+
+
+
+def delete_oldest_records(table_name: str, db_engine: db.Engine, connection: Connection, metadata,
+                          date_cutoff: str = STR_DAY_AGO):
+    """
+    Deletes records older than 24 hours (by attribute 'datetime') from db table with given name
+    (watering/recording).
+    """
+    try:
+        table = db.Table(table_name, metadata, autoload=True,
+                         autoload_with=db_engine)
+        query = db.delete(table).where(table.columns.datetime < str.encode(date_cutoff))
+        connection.execute(query)
+        connection.commit()
+    except Exception as e:
+        raise e
+
+
+def update_rds_and_s3():
+    """
+    Gets any records from watering and recording tables in df older than 24hours, combines with
+    the csv files in s3 bucket for current day (if any exist), and saves the result as csvs in the
+    bucket with a name ending in _{day}.csv.
+    """
+    db_engine = get_database_engine()
+    db_connection = db_engine.connect()
+    db_metadata = db.MetaData(schema=environ['DB_SCHEMA'])
+
+    s3_client = client("s3",
```

Comment:
could put this in a get_s3_client() function here?

Reply:
Personally, because it's a single line and is only used once, I wouldn't